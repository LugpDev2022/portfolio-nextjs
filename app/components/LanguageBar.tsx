import { useMemo } from 'react';
import { getLanguageLevel } from '../helpers/getLanguageLevel';

interface Props {
  language: string;
  percentage: number;
}

const LanguageBar: React.FC<Props> = ({ language, percentage }) => {
  const level = useMemo(() => getLanguageLevel(percentage), [percentage]);

  return (
    <div className='my-4 block w-full text-white sm:my-6 sm:flex'>
      <span className='me-5 block basis-[80px] text-left text-lg font-semibold tracking-wide sm:text-right'>
        {language}
      </span>

      <div className='h-[25px] w-full sm:flex sm:w-10/12'>
        <div
          className='me-2 h-[25px] bg-[#255880]'
          style={{ width: `${percentage}%` }}
        ></div>
        <span className='hidden text-lg sm:inline'>{level}</span>
      </div>
    </div>
  );
};

export default LanguageBar;
