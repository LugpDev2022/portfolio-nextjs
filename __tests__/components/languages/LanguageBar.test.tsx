import LanguageBar from '@/app/components/languages/LanguageBar';
import { render, screen } from '@testing-library/react';

//TODO: Test if the language bar has the given percentage size

describe('tests on <LanguageBar />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('should match snapshot', () => {
    const { container } = render(
      <LanguageBar language='Japanese' percentage={10} />
    );

    expect(container).toMatchSnapshot();
  });

  test('should render the language name', () => {
    const language = 'Japanese';
    render(<LanguageBar language={language} percentage={10} />);

    expect(screen.getByText(language)).toBeTruthy;
  });

  test('should render the language level', () => {
    render(<LanguageBar language='Japanese' percentage={10} />);

    expect(screen.getByText('Beginner')).toBeTruthy();
  });
});
