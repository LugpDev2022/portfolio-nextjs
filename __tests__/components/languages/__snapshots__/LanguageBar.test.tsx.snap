// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <LanguageBar /> should match snapshot 1`] = `
<div>
  <div
    class="my-4 block w-full text-white sm:my-6 sm:flex"
  >
    <span
      class="me-5 block basis-[80px] text-left text-lg font-semibold tracking-wide sm:text-right"
    >
      Japanese
    </span>
    <div
      class="h-[25px] w-full sm:flex sm:w-10/12"
    >
      <div
        class="me-2 h-[25px] bg-[#255880]"
        data-testid="percentageBar"
        style="width: 10%;"
      />
      <span
        class="hidden text-lg sm:inline"
      >
        Beginner
      </span>
    </div>
  </div>
</div>
`;
